rsocket-core/src/main/java/io/rsocket/fragmentation/FragmentationDuplexConnection.java

```diff
@@ -30,139 +39,88 @@
  *     and Reassembly</a>
  */
 public final class FragmentationDuplexConnection implements DuplexConnection {
-  public FragmentationDuplexConnection(DuplexConnection connection, int mtu) {}
-
-  @Override
-  public Mono<Void> send(Publisher<ByteBuf> frames) {
-    return null;
-  }
-
-  @Override
-  public Flux<ByteBuf> receive() {
-    return null;
-  }
-
-  @Override
-  public Mono<Void> onClose() {
-    return null;
-  }
-
-  @Override
-  public void dispose() {}
-
-  /*
-  private final ByteBufAllocator byteBufAllocator;
-
+  private static final int MIN_MTU_SIZE = 64;
+  private static final Logger logger = LoggerFactory.getLogger(FragmentationDuplexConnection.class);
   private final DuplexConnection delegate;
+  private final int mtu;
+  private final ByteBufAllocator allocator;
+  private final FrameReassembler frameReassembler;
+  private final boolean encodeLength;
 
-  private final FrameFragmenter frameFragmenter;
-
-  private final IntObjectHashMap<FrameReassembler> frameReassemblers = new IntObjectHashMap<>();
-
-  */
-  /**
-   * Creates a new instance.
-   *
-   * @param delegate the {@link DuplexConnection} to decorate
-   * @param maxFragmentSize the maximum fragment size
-   * @throws NullPointerException if {@code delegate} is {@code null}
-   * @throws IllegalArgumentException if {@code maxFragmentSize} is not {@code positive}
-   */
-  /*
-  // TODO: Remove once ByteBufAllocators are shared
-  public FragmentationDuplexConnection(DuplexConnection delegate, int maxFragmentSize) {
-    this(PooledByteBufAllocator.DEFAULT, delegate, maxFragmentSize);
-  }
-
-  */
-  /**
-   * Creates a new instance.
-   *
-   * @param byteBufAllocator the {@link ByteBufAllocator} to use
-   * @param delegate the {@link DuplexConnection} to decorate
-   * @param maxFragmentSize the maximum fragment size. A value of 0 indicates that frames should not
-   *     be fragmented.
-   * @throws NullPointerException if {@code byteBufAllocator} or {@code delegate} are {@code null}
-   * @throws IllegalArgumentException if {@code maxFragmentSize} is not {@code positive}
-   */
-  /*
   public FragmentationDuplexConnection(
-      ByteBufAllocator byteBufAllocator, DuplexConnection delegate, int maxFragmentSize) {
-
-    this.byteBufAllocator =
-        Objects.requireNonNull(byteBufAllocator, "byteBufAllocator must not be null");
-    this.delegate = Objects.requireNonNull(delegate, "delegate must not be null");
-
-    NumberUtils.requireNonNegative(maxFragmentSize, "maxFragmentSize must be positive");
+      DuplexConnection delegate, ByteBufAllocator allocator, int mtu, boolean encodeLength) {
+    Objects.requireNonNull(delegate, "delegate must not be null");
+    Objects.requireNonNull(allocator, "byteBufAllocator must not be null");
+    if (mtu < MIN_MTU_SIZE) {
+      throw new IllegalArgumentException("smallest allowed mtu size is 64 bytes");
```

Comment:
for a non hot path exception, worth referring to MIN_MTU_SIZE in the message instead of 64